docs/blog/posts/pydantic.md

```diff
@@ -0,0 +1,235 @@
+                                    "name": {"title": "Name", "type": "string"},
+                                    "author": {"title": "Author", "type": "string"},
+                                },
+                                "required": ["name", "author"],
+                                "title": "Package",
+                                "type": "object",
+                            }
+                        },
+                        "properties": {
+                            "packages": {
+                                "items": {"$ref": "#/$defs/Package"},
+                                "title": "Packages",
+                                "type": "array",
+                            }
+                        },
+                        "required": ["packages"],
+                        "title": "Packages",
+                        "type": "object",
+                    },
+                },
+            }
+        ],
+        tool_choice={
+            "type": "function",
+            "function": {"name": "Requirements"},
+        },
+    )
+
+    resp = json.loads(resp.choices[0].message.tool_calls[0].function.arguments)
+    ```
+
+Now, notice in this example that the prompts we use contain purely the data we want, where the tools and tools choice now capture the schemas we want to output. This speration of concerns makes it much easier organize the 'data' and the 'description' of the data that we want back out.
+
+```python
+from typing import List
+from pydantic import BaseModel
+from openai import OpenAI
+
+client = OpenAI()
+
+
+class Package(BaseModel):
+    name: str
+    author: str
+
+
+class Packages(BaseModel):
+    packages: List[Package]
+
+
+resp = client.chat.completions.create(
+    model="gpt-3.5-turbo",
+    messages=[
+        {
+            "role": "user",
+            "content": "Pydantic and FastAPI?",
+        },
+    ],
+    tools=[
+        {
+            "type": "function",
+            "function": {
+                "name": "Requirements",
+                "description": "A list of packages and their first authors.",
+                "parameters": Packages.model_json_schema(),
+            },
+        }
+    ],
+    tool_choice={
+        "type": "function",
+        "function": {"name": "Requirements"},
+    },
+)
+
+Packages.model_validate_json(
+    resp.choices[0].message.tool_calls[0].function.arguments
+)
+```
+
+```json
+{
+  "packages": [
+    {
+      "name": "pydantic",
+      "author": "Samuel Colvin"
+    },
+    {
+      "name": "fastapi",
+      "author": "Sebastián Ramírez"
+    }
+  ]
+}
+```
+
+## Instructor
+
+Although this example may seem contrived, it demonstrates how Pydantic can be used to generate structured data from language models. To simplify this pattern, I have developed a small library called `instructor` that patches the `OpenAI` client. This library offers convenient features such as JSON mode, function calling, tool usage, and open source models.
```

Comment:
"Although this example may seem contrived" - no need to sell yourself short - it's a good example :).